### PR TITLE
simplifier pass CI: uncomment passing CI job

### PR DESCRIPTION
**Pull request by gxyd**

### CODE REVIEW

File: src/libasr/pass/simplifier.cpp [hunk 1]
```diff
@@ -1838,12 +1838,12 @@ class VerifySimplifierASROutput:
     private:
 
     Allocator& al;
-    ExprsWithTargetType& exprs_with_target;
+    // ExprsWithTargetType& exprs_with_target;
```
Comment (gxyd) [inline src/libasr/pass/simplifier.cpp:1841]: Seems like this field is being used, and probably not getting detected because of use in a macro.

Comment (certik) [reply src/libasr/pass/simplifier.cpp:1841]: You can put `[[maybe_unused]]` in front I think.

Comment (gxyd) [reply src/libasr/pass/simplifier.cpp]: Yes, I think that's a good simple solution to it.

**Comment by gxyd** [comment]
If I used `[[maybe_unused]]` then the *Check Debug build* job fails with the below error:
```bash
sccache /usr/bin/c++ -DHAVE_TARGET_X86=1 -DHAVE_WHEREAMI=1 -DHAVE_ZLIB=1 -DLCOMPILERS_FAST_ALLOC=1 -I/home/runner/work/lfortran/lfortran/src/libasr/.. -isystem /home/runner/micromamba/envs/lf/include -Werror -Wall -Wextra -g -ggdb -std=gnu++17 -fPIC -fno-rtti -MD -MT src/libasr/CMakeFiles/asr.dir/pass/simplifier.cpp.o -MF src/libasr/CMakeFiles/asr.dir/pass/simplifier.cpp.o.d -o src/libasr/CMakeFiles/asr.dir/pass/simplifier.cpp.o -c /home/runner/work/lfortran/lfortran/src/libasr/pass/simplifier.cpp
/home/runner/work/lfortran/lfortran/src/libasr/pass/simplifier.cpp:1841:43: error: ‘maybe_unused’ attribute ignored [-Werror=attributes]
 1841 |     [[maybe_unused]] ExprsWithTargetType& exprs_with_target;

```

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -563,10 +563,10 @@ jobs:
...
             git checkout fcde66ca86348eb0c4012dbdf0f4d8dba61261d8
             $(pwd)/../src/bin/lfortran ./src/minpack.f90 -c --legacy-array-sections
-      #       $(pwd)/../src/bin/lfortran ./examples/example_hybrd.f90 --legacy-array-sections
-      #       $(pwd)/../src/bin/lfortran ./examples/example_hybrd1.f90 --legacy-array-sections
-      #       $(pwd)/../src/bin/lfortran ./examples/example_lmdif1.f90 --legacy-array-sections
-      #       $(pwd)/../src/bin/lfortran ./examples/example_lmder1.f90 --legacy-array-sections
+            $(pwd)/../src/bin/lfortran ./examples/example_hybrd.f90 --legacy-array-sections
+            $(pwd)/../src/bin/lfortran ./examples/example_hybrd1.f90 --legacy-array-sections
+            $(pwd)/../src/bin/lfortran ./examples/example_lmdif1.f90 --legacy-array-sections
+            $(pwd)/../src/bin/lfortran ./examples/example_lmder1.f90 --legacy-array-sections
```
Comment (kmr-srbh) [inline .github/workflows/CI.yml:569]: @gxyd could you please confirm if the output of this is the same as the one produced by GFortran? :)

Comment (gxyd) [reply .github/workflows/CI.yml:569]: I didn't compare it with GFortran, but compared it with the current *main*, and the output seems to be same. I think that should be enough.

Comment (Pranavchiku) [inline .github/workflows/CI.yml:569]: ouch, I applied same changes via https://github.com/lfortran/lfortran/pull/4841

**Review by Pranavchiku — APPROVED** [review]
I am sorry I didn't check this. I rebased it, this PR looks good to me.

**Pull request merged**